api/src/main/java/org/apache/iceberg/util/ExceptionUtil.java

```diff
@@ -113,15 +114,20 @@ public static <R, E1 extends Exception, E2 extends Exception, E3 extends Excepti
         try {
           finallyBlock.run();
         } catch (Exception e) {
-          LOG.warn("Suppressing failure in finally block", e);
           if (failure != null) {
+            LOG.warn("Suppressing failure in finally block", e);
             failure.addSuppressed(e);
+            tryThrowAs(failure, e1Class);
```

Comment:
This is probably not necessary and actually goes against the ErrorProne guideline of not throwing from `finally`.

The only way `failure` can be non-null here is if it was thrown from the main `try` block, in which case the generated bytecode will ensure it is re-thrown when `finally` completes.

Reply:
In this case, `failure` is already thrown because it is thrown in the `catch` block. `tryThrowAs` needs to be removed here. The logic was correct before, except for the wrong exception in the next block.

Reply:
TIL. Will remove. Thanks guys.

Reply:
Oh actually yeah I understand. The `failure` would have already been thrown, just the finally block gets run and we add one more suppression to it. So we don't need to throw it again. Thanks for the explanation.